src/main/java/com/boot/service/RegistrationService.java

```diff
@@ -0,0 +1,72 @@
+package com.boot.service;
+
+
+import com.boot.entity.Role;
+import com.boot.entity.User;
+import com.boot.pojo.MessageResponse;
+import com.boot.pojo.RegistrationRequest;
+import com.boot.repository.RoleRepository;
+import com.boot.repository.UserRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+
+import java.util.HashSet;
+import java.util.Set;
+
+@Service
+public class RegistrationService {
+
+    @Autowired
+    private UserRepository userRepository;
+
+    @Autowired
+    private RoleRepository roleRepository;
+
+    @Autowired
+    private PasswordEncoder passwordEncoder;
+
+
+    public ResponseEntity<?> registrationUser(RegistrationRequest registrationRequest) {
+
+        if (userRepository.existsByUsername(registrationRequest.getUsername())) {
+            return ResponseEntity
+                    .badRequest()
+                    .body(new MessageResponse("Error: Username is exist"));
+        }
+
+        if (userRepository.existsByEmail(registrationRequest.getEmail())) {
+            return ResponseEntity
+                    .badRequest()
+                    .body(new MessageResponse("Error: Email is exist"));
+        }
+
+        Set<Role> roles = new HashSet<>();
+
+        switch (registrationRequest.getRole()) {
+            case "admin":
```

Comment:
finding from repository you can move to private method. So that code might looks like : if("admin".equals(registrationRequest.getRole))
 roles.add(getRole(Role.Name.ROLE_ADMIN) 
    else roles.add(getRole(Role.Name.ROLE_USER)

Reply:
Logic with roles might me moved to separate private method : List<Role> roles = getRoles(request);